src/entrypoints/accounts-actions/execute.ts

```diff
@@ -61,35 +76,109 @@ export async function populateExecuteEnqueue(
   transaction: TransactionRequest,
   sign: SignTypedDataCallback
 ): Promise<TransactionRequest> {
-  account = getAddress(account);
-  salt = salt || saltFromTimestamp();
+  const { domain, primaryType, types, message } = generateTypedData(
+    { account, chainId, salt },
+    transaction
+  );
 
-  const delayMod = {
-    address: predictDelayModAddress(account),
-    iface: deployments.delayModMastercopy.iface,
+  const signature = await sign({ domain, primaryType, types, message });
+
+  const txRequest = getTransactionRequest({
+    account,
+    transaction,
+    message,
+    signature,
+  });
+
+  return txRequest;
+}
+
+/**
+ * Constructs the final transaction request for executing a queued transaction
+ * on the Delay Modifier. This function combines the encoded transaction data
+ * with the salt and signature to create a relay-ready transaction.
+ *
+ * @param parameters - Object containing account, transaction, message, and signature
+ * @param parameters.account - The address of the account Safe
+ * @param parameters.transaction - The original transaction to be executed
+ * @param parameters.message - The message object containing salt and data from typed data generation
+ * @param parameters.signature - The EIP-712 signature from the account owner
+ * @returns The complete transaction request ready for relay execution
+ *
+ * @example
+ * import { getTransactionRequest } from "@gnosispay/account-kit";
+ *
+ * const txRequest = getTransactionRequest({
+ *   account: "0x<address>",
+ *   transaction: { to: "0x<address>", value: 0n, data: "0x<bytes>" },
+ *   message: { salt: "0x<bytes32>", data: "0x<encoded_data>" },
+ *   signature: "0x<signature>"
+ * });
+ */
+export const getTransactionRequest = ({
```

Comment:
I think we should probably change the signature of `populateExecuteEnqueue` to `getSignedTransactionRequest`.

Right now we have the following:
- `populateExecuteEnqueue`
- `getTransactionRequest`

I think for a consumer of the API it can be confusing to decide which one to use. Also, "populate" does not really populate anything, it signs a delay transaction ready to be executed.

Not directly related to your PR, but I think we should review naming with these new changes.